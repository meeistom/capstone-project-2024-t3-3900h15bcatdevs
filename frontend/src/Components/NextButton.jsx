import React from 'react';
import { Button } from 'react-bootstrap';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faArrowRight } from '@fortawesome/free-solid-svg-icons';

import 'bootstrap/dist/css/bootstrap.min.css';
import '../index.css';

export { NextButton };

function NextButton({ id, onClick, disabled, children }) {
  return (
    <Button
      id={id}
      variant="primary"
      size="lg"
      aria-label="next-button"
      onClick={onClick}
      disabled={disabled}>
      <span className="me-2">{children}</span>
      <FontAwesomeIcon icon={faArrowRight} />
    </Button>
  );
}
