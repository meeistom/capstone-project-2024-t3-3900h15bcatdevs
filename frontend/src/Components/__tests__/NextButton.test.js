import { render, screen, fireEvent, cleanup } from "@testing-library/react";
import { NextButton } from "../NextButton";
import renderer from "react-test-renderer";
import "@testing-library/jest-dom";

afterEach(() => {
  cleanup();
});

test("should render next button", () => {
  render(<NextButton onClick={null}>Hello</NextButton>);
  const nextButton = screen.getByRole("button", { name: "next-button" });

  expect(nextButton).toBeInTheDocument();
});

test("should match snapshot", () => {
  const tree = renderer
    .create(<NextButton onClick={null}>Hello</NextButton>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("should pass down children", () => {
  render(<NextButton onClick={null}>Hello</NextButton>);
  const nextButton = screen.getByRole("button", { name: "next-button" });

  expect(nextButton).toHaveTextContent("Hello");
});

test("should handle onClick", () => {
  const onClick = jest.fn();

  render(<NextButton onClick={onClick}>Hello</NextButton>);
  const nextButton = screen.getByRole("button", { name: "next-button" });

  fireEvent.click(nextButton);
  expect(onClick).toHaveBeenCalledTimes(1);
});
