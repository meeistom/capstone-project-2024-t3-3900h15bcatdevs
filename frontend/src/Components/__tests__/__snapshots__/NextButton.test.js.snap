// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<button
  aria-label="next-button"
  className="btn btn-primary btn-lg"
  disabled={false}
  onClick={null}
  type="button"
>
  <span
    className="me-2"
  >
    Hello
  </span>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-arrow-right "
    data-icon="arrow-right"
    data-prefix="fas"
    focusable="false"
    role="img"
    style={Object {}}
    viewBox="0 0 448 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M438.6 278.6c12.5-12.5 12.5-32.8 0-45.3l-160-160c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L338.8 224 32 224c-17.7 0-32 14.3-32 32s14.3 32 32 32l306.7 0L233.4 393.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0l160-160z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
</button>
`;
